control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -2051,6 +2071,88 @@ func (r *reconciler) ensureGuestAdmissionWebhooksAreValid(ctx context.Context) e
 	return errors.NewAggregate(errs)
 }
 
+func (r *reconciler) reconcileKubeletConfig(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	wantCMList := &corev1.ConfigMapList{}
+	opts := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{nodepool.KubeletConfigConfigMapLabel: "true"}),
+		Namespace:     r.hcpNamespace,
+	}
+	if err := r.cpClient.List(ctx, wantCMList, opts); err != nil {
+		return fmt.Errorf("failed to list KubeletConfig ConfigMaps from namespace %s: %w", r.hcpNamespace, err)
+	}
+
+	haveCMList := &corev1.ConfigMapList{}
+	opts = &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{nodepool.KubeletConfigConfigMapLabel: "true"}),
+		Namespace:     ConfigManagedNamespace,
+	}
+	if err := r.client.List(ctx, haveCMList, opts); err != nil {
+		return fmt.Errorf("failed to list KubeletConfig ConfigMaps: %w", err)
+	}
+
+	want := set.Set[string]{}
+	for _, cm := range wantCMList.Items {
+		want.Insert(cm.Name)
+	}
+
+	have := set.Set[string]{}
+	for _, cm := range haveCMList.Items {
+		have.Insert(cm.Name)
+	}
+
+	toCreate := want.Difference(have)
+	if len(toCreate) > 0 {
+		log.WithValues("toCreate", toCreate.UnsortedList())
+	}
+	toDelete := have.Difference(want)
```

Comment:
Hey @jparrill ,

Thank you for your question.
So assuming we have `ConfigMapA` and `ConfigMapB`.
Now the user add `ConfigMapC` and remove `ConfigMapB`

If HCCO only adds what `want` source has, then you'll end up with `ConfigMapA`, `ConfigMapB` and `ConfigMapC` in the cluster.

Please keepl me honest here, but your comment seems to be true only if the `updateStrategy` of the nodes is `replace`, but in case of `inPlace` it's not.

Reply:
I don't see how the upgrade type is relevant. This objects are persisted in etcd. You need to account for deleting the ones there's no intent for either way replace/inplace.

Reply:
That is what this code is doing - deleting the ones which are no longer relevant